### ISSUE TITLE
[BUG] When using the ShenYu Monitor plugin to monitor Http request QPS, TPS and other related metrics, the gateway service throws a NullPointerException

### ISSUE THREAD

**Issue by wangdeng-wd** [apache/shenyu#4630]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

version:2.4.0

When using the ShenYu Monitor plugin to monitor Http request QPS, TPS and other related metrics, the gateway service throws a NullPointerException

### Expected Behavior

1.Call the example-grpc service and return normally
2.The metrics information includes shenyu_http_request_total, shenyu_request_total and other indicators

### Steps To Reproduce

0. Start admin, bootstrap, example-grpc services
1. Open Monitor Plugin and configure metrics information  【{"metricsName":"prometheus","host":"localhost","port":"9190","async":"true"}】
![image](https://github.com/apache/shenyu/assets/52487228/9cbb0701-0f26-48b3-86a1-42ae14835d97)
2. Selector and rule configure full traffic rules in the Monitor Plugin 
![image](https://github.com/apache/shenyu/assets/52487228/e6baa820-65c9-4ba3-9408-26592929ac0d)

3. Call the example-grpc service through the gateway 
```
curl --location --request POST 'http://localhost:9195/grpc/serverStreamingFun' \
--header 'Content-Type: application/json' \
--data-raw '{
    "data": [
        {
            "message": "hello grpc"
        }
    ]
}'
```
4. bootstrap NullPointerException
```
2023-05-11 19:36:50 [shenyu-netty-nio-3] ERROR org.apache.shenyu.web.handler.GlobalErrorHandler - [28e32ca6] Resolved [NullPointerException: null] for HTTP POST /grpc/serverStreamingFun
2023-05-11 19:36:50 [shenyu-netty-nio-3] ERROR org.springframework.boot.autoconfigure.web.reactive.error.AbstractErrorWebExceptionHandler - [28e32ca6]  500 Server Error for HTTP POST "/grpc/serverStreamingFun"
java.lang.NullPointerException: null
	at org.apache.shenyu.metrics.prometheus.register.PrometheusMetricsRegister.counterIncrement(PrometheusMetricsRegister.java:82)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.apache.shenyu.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.FileSizeFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.bootstrap.filter.HealthFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.CrossFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP POST "/grpc/serverStreamingFun" [ExceptionHandlingWebHandler]
Stack trace:
		at org.apache.shenyu.metrics.prometheus.register.PrometheusMetricsRegister.counterIncrement(PrometheusMetricsRegister.java:82)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.lambda$counterIncrement$3(MetricsReporter.java:156)
		at java.util.Optional.ifPresent(Optional.java:159)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.counterIncrement(MetricsReporter.java:156)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.counterIncrement(MetricsReporter.java:145)
		at org.apache.shenyu.plugin.monitor.MonitorPlugin.doExecute(MonitorPlugin.java:50)
		at org.apache.shenyu.plugin.base.AbstractShenyuPlugin.execute(AbstractShenyuPlugin.java:93)
		at org.apache.shenyu.web.handler.ShenyuWebHandler$DefaultShenyuPluginChain.lambda$execute$0(ShenyuWebHandler.java:109)
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44)
    	................................
    	........................................
		at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1050)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:750)
```

### Environment

```markdown
ShenYu version(s):2.4.0
```

### Debug logs

2023-05-11 19:36:50 [shenyu-netty-nio-3] ERROR org.apache.shenyu.web.handler.GlobalErrorHandler - [28e32ca6] Resolved [NullPointerException: null] for HTTP POST /grpc/serverStreamingFun
2023-05-11 19:36:50 [shenyu-netty-nio-3] ERROR org.springframework.boot.autoconfigure.web.reactive.error.AbstractErrorWebExceptionHandler - [28e32ca6]  500 Server Error for HTTP POST "/grpc/serverStreamingFun"
java.lang.NullPointerException: null
	at org.apache.shenyu.metrics.prometheus.register.PrometheusMetricsRegister.counterIncrement(PrometheusMetricsRegister.java:82)
	Suppressed: reactor.core.publisher.FluxOnAssembly$OnAssemblyException: 
Error has been observed at the following site(s):
	|_ checkpoint ⇢ org.apache.shenyu.web.configuration.ErrorHandlerConfiguration$1 [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.WebSocketParamFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.FileSizeFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.bootstrap.filter.HealthFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.apache.shenyu.web.filter.CrossFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ org.springframework.boot.actuate.metrics.web.reactive.server.MetricsWebFilter [DefaultWebFilterChain]
	|_ checkpoint ⇢ HTTP POST "/grpc/serverStreamingFun" [ExceptionHandlingWebHandler]
Stack trace:
		at org.apache.shenyu.metrics.prometheus.register.PrometheusMetricsRegister.counterIncrement(PrometheusMetricsRegister.java:82)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.lambda$counterIncrement$3(MetricsReporter.java:156)
		at java.util.Optional.ifPresent(Optional.java:159)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.counterIncrement(MetricsReporter.java:156)
		at org.apache.shenyu.metrics.reporter.MetricsReporter.counterIncrement(MetricsReporter.java:145)
		at org.apache.shenyu.plugin.monitor.MonitorPlugin.doExecute(MonitorPlugin.java:50)
		at org.apache.shenyu.plugin.base.AbstractShenyuPlugin.execute(AbstractShenyuPlugin.java:93)
		at org.apache.shenyu.web.handler.ShenyuWebHandler$DefaultShenyuPluginChain.lambda$execute$0(ShenyuWebHandler.java:109)
		at reactor.core.publisher.MonoDefer.subscribe(MonoDefer.java:44)
    	................................
    	........................................
		at io.netty.util.concurrent.SingleThreadEventExecutor$6.run(SingleThreadEventExecutor.java:1050)
		at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
		at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
		at java.lang.Thread.run(Thread.java:750)

**Comment by yu199195** [comment]
it be fixed，maybe upgrade to 2.5.x~

**Comment by wangdeng-wd** [comment]
2.5.x has fixed
